### ISSUE TITLE
M5 — Normalisation layers

### ISSUE THREAD

**Issue by harshitagrawal2O** [harshitagrawal2O/tensorkit#5]
**Milestone 5 of 8** — see [`SPEC.md`](../blob/main/SPEC.md) section 4 for the full definition of done.

**CS subject:** Deep Learning (normalisation statistics, train/eval divergence)

**Nobody is working on this.** Comment to claim it. One person per milestone at a time; if a
claim goes quiet for two weeks it is fair game again.

## Scope

`LayerNorm`, `BatchNorm1d`, `BatchNorm2d`, including running buffers.

## Definition of done

- All layers pass float64 gradcheck
- Running statistics are **buffers, not parameters** — the optimiser must not reach them
- In `eval()`, an item's output does not depend on the rest of the batch
- Batch size 1 in `train()` raises rather than dividing by zero

## Where it gets interesting

The backward pass has **three terms**, not one, because μ and σ² each depend on every element of the reduction group. Drop an indirect term and you are wrong by a factor that shrinks as the batch grows — about 0.4% at 256, which trains almost normally.

## Ground rules

- Do **not** weaken a test to make it pass. If you believe a test is wrong, open a
  [spec-defect issue](../issues/new?template=spec-defect.yml) instead — several tests have
  already been corrected that way.
- Keep the existing docstrings; replace only the `raise NotImplementedError(...)` line.
- `ruff check`, `ruff format --check` and `mypy` must be clean.

Chapter-length background is in [the book](../blob/main/docs/tensorkit-book.pdf).

```bash
python -m pytest -m m5 -q
```
